### DISCUSSION TITLE
How can I develop code-server ??

### DISCUSSION THREAD

**Discussion by chen850717** [coder/code-server#3059] [Q&A]
If I want to change code in code-server, which IDE can I use
I had try Electron to open the project, but it's failed
I attempt to change code-server in vs code but some file was mess like this

```
var define,AMDLoader,CSSLoaderPlugin,_amdLoaderGlobal=this,_commonjsGlobal="object"==typeof global?global:{};!function(e){e.global=_amdLoaderGlobal;var t=function(){function t(){this._detected=!1,this._isWindows=!1,this._isNode=!1,this._isElectronRenderer=!1,this._isWebWorker=!1}return Object.defineProperty(t.prototype,"isWindows",{get:function(){return this._detect(),this._isWindows},enumerable:!0,configurable:!0}),Object.defineProperty(t.prototype,"isNode",{get:function(){return this._detect(),this._isNode},enumerable:!0,configurable:!0}),Object.defineProperty(t.prototype,"isElectronRenderer",{get:function(){return this._detect(),this._isElectronRenderer},enumerable:!0,configurable:!0}),Object.defineProperty(t.prototype,"isWebWorker",{get:function(){return this._detect(),this._isWebWorker},enumerable:!0,configurable:!0}),t.prototype._detect=function(){this._detected||(this._detected=!0,this._isWindows=t._isWindows(),this._isNode="undefined"!=typeof module&&!!module.exports,
this._isElectronRenderer="undefined"!
```

**Comment by jsjoeio** [comment]
Sorry to hear you're having trouble. Have you followed the [`CONTRIBUTING.md`](https://github.com/cdr/code-server/blob/main/docs/CONTRIBUTING.md#requirements)? And are you able to run code-server locally using `yarn watch`?

**Reply by chen850717** [reply]
I try yarn watch but it not work
`$ VSCODE_IPC_HOOK_CLI= NODE_OPTIONS=--max_old_space_size=32384 ts-node ./ci/dev/watch.ts
'VSCODE_IPC_HOOK_CLI' 
It is not an internal or external command, executable program or batch file.
error Command failed with exit code 1.`

**Reply by jsjoeio** [reply]
Hmm...I think we'll need more information to help you get this figured out. Sorry for the trouble!

Do you mind opening up a [bug report](https://github.com/cdr/code-server/issues/new?assignees=&labels=&template=bug-report.md&title=)?

**Comment by trasherdk** [comment]
Set `VSCODE_IPC_HOOK_CLI` to something. An empty string will probably do `VSCODE_IPC_HOOK_CLI=""`
